src/test/CMakeLists.txt

```diff
@@ -678,6 +678,30 @@ add_executable(unittest_bufferlist
 add_ceph_unittest(unittest_bufferlist)
 target_link_libraries(unittest_bufferlist global)
 
+# compiletest_cxx11_client
+add_executable(compiletest_cxx11_client
+  cxx11_client.cc
+  )
+if(CMAKE_VERSION VERSION_LESS "3.8")
+  # this is ugly as we'll end with -std=c++11 overriding the previous -std=c++17
+  # I would love to have a better way for old Cmakes
+  set_target_properties(compiletest_cxx11_client
+    PROPERTIES COMPILE_FLAGS "-std=c++11 -Werror -pedantic"
+    )
+else()
+  set_target_properties(compiletest_cxx11_client
+    CMAKE_CXX_STANDARD 11
+    CXX_STANDARD_REQUIRED ON
+    )
+  set_target_properties(compiletest_cxx11_client
+    PROPERTIES COMPILE_FLAGS "-Werror -pedantic"
+    )
+endif()
+add_ceph_test(compiletest_cxx11_client
```

Comment:
Alternatively we could enroll `../qa/workunits/rados/test_librados_build.sh` here. This would require a little tuning to e.g. pass `-L` to let gcc find the libs and maybe `-pedantic`. However, the things I find useful in the current approach are:
* having the explicit list of our public headers and
* being focused solely on the C++11 part -- no cluster is necessary for validation.